test/puppet/puppet_class_test.rb

```diff
@@ -42,4 +38,31 @@ def test_scan_directory_loads_eparser_scanner
     Proxy::Puppet::ClassScannerEParser.expects(:scan_directory).with('/foo',  'development')
     Proxy::Puppet::PuppetClass.scan_directory('/foo', 'development', true)
   end
+
+  def test_json_serialization
+    clazz = Proxy::Puppet::PuppetClass.new("foreman_proxy::install",
+                                   "namedconf_path"=>"${::dns::params::namedconf_path}",
+                                   "dnsdir"=>"${::dns::params::dnsdir}")
+
+    assert clazz.to_json.include?("\"json_class\":\"Proxy::Puppet::PuppetClass\"")
+    assert clazz.to_json.include?("\"klass\":\"foreman_proxy::install\"")
+    assert(
+      clazz.to_json.include?(
+          "\"params\":{\"namedconf_path\":\"${::dns::params::namedconf_path}\",\"dnsdir\":\"${::dns::params::dnsdir}\"")
```

Comment:
This is failing intermittently on 1.8.7.. I'd probably bet that the keys are swapping order as the iteration order isn't deterministic under 1.8.